### PR TITLE
[TS-347] Comparison Tool for My Dashboard and Other Dashboard (without graph)

### PR DESCRIPTION
**Pull request by AlyaNaseer**
**EDIT: added button to dashboard now so just check it out there**

Navigate to dashboard
Click "Compare Returns" on Returns graph

image
Here we see the 5 users, and here it is in the code (pages/ComparisonTool/ComparisonTool.vue):
image

If we change it to 1 user:
image
image

Here are two videos showing the functionality of the modal (each option except for Maximum returns the date range in the console). By default the custom "to" and "from" dates are disabled unless the custom option is selected. Pops up a calendar to choose dates and I added date validation (can't have a "to" date be before the "from" date). 

https://user-images.githubusercontent.com/46854444/158444544-ff049184-a40e-4dd8-81bb-630903605442.mp4
https://user-images.githubusercontent.com/46854444/158444551-dac2c81f-29d9-431e-8a7d-5ab67dad4276.mp4

Cypress tests are also done.

The way the custom dates works is as follows:
- if no dates are entered, by default show today's returns
- if "from" date is entered but not "to" date, show date range from "from" date to today
- if "to" date is entered but not "from" date, show all data from beginning to date specified (by default beginning is 1900-01-01)

### CODE REVIEW

**Review by bowe99 — CHANGES_REQUESTED** [review]
Is there a way we could restrict keyboard input on the text boxes because I think I found a way around your constraints 

https://user-images.githubusercontent.com/50396173/158460469-560a558c-20ab-408b-b0b7-e6f63d5704da.mp4

**Comment by AlyaNaseer** [comment]
> Is there a way we could restrict keyboard input on the text boxes because I think I found a way around your constraints
> 
>  Bug_Trim.mp4

Nice catch, I'll look into that

**Review by bowe99 — DISMISSED** [review]
This is great work! Works perfectly now

File: src/components/ComparisonTool/ComparisonModal.vue [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+          v-bind="attrs"
+          data-cy="modal-activate"
+          outlined
+          small
+          v-on="on"
+        >
+          COMPARE RETURNS
+        </v-btn>
+      </template>
+
```
Comment (itsShaVawn) [inline src/components/ComparisonTool/ComparisonModal.vue:21]: no need for the extra space here

File: src/components/ComparisonTool/ComparisonModal.vue [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            <v-card
+              height="500"
+              width="100%"
+              class="d-flex justify-center align-center"
+            >
+              Insert graph here, and add "flat" to this v-card
+            </v-card>
+          </v-col>
+          <v-col cols="3">
+            <v-card
```
Comment (itsShaVawn) [inline src/components/ComparisonTool/ComparisonModal.vue:50]: this can be a one liner

File: src/components/ComparisonTool/ComparisonModal.vue [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+          </v-col>
+        </v-row>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
```
Comment (itsShaVawn) [inline src/components/ComparisonTool/ComparisonModal.vue:167]: add an export name

**Review by itsShaVawn — CHANGES_REQUESTED** [review]
image

Just some small things:

- I would make the button the same style as the "New Item" button for consistency
- Maybe add a bit more margin above the button, it seems a bit close to the container to me
- What exactly does the refresh button do? Or what is it supposed to do? Does it refresh the graph or refresh the filters?

Great work!

**Comment by itsShaVawn** [comment]
image

I think also for small screens you could display only the colored dots or only dots and first name so it looks cleaner

**Comment by itsShaVawn** [comment]
image
image

Maybe some adjustments here for small screens. We can't see the whole calendar or any date that is inputted

**Comment by itsShaVawn** [comment]
> image
> 
> I think also for small screens you could display only the colored dots or only dots and first name so it looks cleaner

Just another suggestion here, since on this size, the box for the graph is higher than it is wider, this will make the graph look weird. Maybe try bringing the filters and stuff under the graph card. That way the graph would have more space on smaller screens and your filters would also have more width.

**Review by itsShaVawn — APPROVED** [review]
Nice work! 👍🏼

**Review by bowe99 — APPROVED** [review]

**Pull request merged**